fairscale/utils/testing.py

```diff
@@ -143,25 +149,34 @@ def get_world_sizes() -> List[int]:
 def spawn_for_all_world_sizes(test_func: Callable, world_sizes: List[int] = get_world_sizes(), args: Any = []) -> None:
 
     for world_size in world_sizes:
-        filename = tempfile.mkstemp()[1]
-        context = mp.spawn(test_func, args=(world_size, filename, *args), nprocs=world_size, join=False)  # type: ignore
-        context.join(timeout=60.0)
+        _, filename = tempfile.mkstemp()
+        _, filename_rpc = tempfile.mkstemp()
```

Comment:
for the record, this was the root cause for the hang, rpc needs a unique (different from torch dist) file to be able to initialize file-based

Reply:
wow, impressive debugging!